Container restart loop after `docker compose down/up` of bootstrapped cluster: `git fetch` without auth fails, set -e exits, Docker loops

## Symptoms

After successfully bootstrapping a cluster with the wizard flow, doing \`docker compose down && docker compose up -d\` puts the orchestrator container into a tight restart loop:

\`\`\`
[2026-05-12 20:55:11] [orchestrator] Starting orchestrator setup...
[2026-05-12 20:55:11] [setup-credentials] WARNING: GH_TOKEN not set
[2026-05-12 20:55:11] [orchestrator] Project repo already cloned, pulling latest...
fatal: could not read Username for 'https://github.com': No such device or address
[2026-05-12 20:55:12] [orchestrator] Starting orchestrator setup...
[2026-05-12 20:55:12] [setup-credentials] WARNING: GH_TOKEN not set
[2026-05-12 20:55:12] [orchestrator] Project repo already cloned, pulling latest...
fatal: could not read Username for 'https://github.com': No such device or address
\`\`\`

Status: \`Restarting\`, looping every ~1s.

## Root cause

[\`.devcontainer/generacy/scripts/resolve-workspace.sh:57-62\`](.devcontainer/generacy/scripts/resolve-workspace.sh#L57-L62):

\`\`\`bash
elif [ -d "${WORKSPACE_DIR}/.git" ]; then
    log "Project repo already cloned, pulling latest..."
    cd "${WORKSPACE_DIR}"
    git fetch origin                                                  # ← no || true
    git pull --ff-only origin "${REPO_BRANCH:-main}" 2>/dev/null || true
fi
\`\`\`

\`git fetch origin\` has no \`|| true\` guard. When \`GH_TOKEN\` is empty (which it is on restart — see below), \`git fetch\` exits non-zero. Combined with \`set -e\` in [\`entrypoint-orchestrator.sh:3\`](.devcontainer/generacy/scripts/entrypoint-orchestrator.sh#L3), the entrypoint terminates. The compose file has \`restart: unless-stopped\`, so Docker immediately restarts the container. The same path runs again. Loop.

## Why \`GH_TOKEN\` is empty on restart

Per generacy-ai/generacy#589 / cluster-base#28, the wizard's credential delivery writes \`/var/lib/generacy/wizard-credentials.env\` (containing \`GH_TOKEN=ghs_...\`) and \`entrypoint-post-activation.sh\` sources it before \`setup-credentials.sh\` runs.

But \`entrypoint-orchestrator.sh\` runs **every** container start — including restarts of an already-bootstrapped cluster. It does NOT source the wizard env file. So on restart:

- The workspace exists from the previous run (sourced from the named \`workspace\` volume)
- \`resolve-workspace.sh\` hits the "Project repo already cloned, pulling latest..." branch
- No \`GH_TOKEN\` in env → git auth fails → \`set -e\` exits

On a **first** activation (fresh cluster, no workspace yet), the flow is fine: clone is deferred to wizard mode, no \`git fetch\` runs in the orchestrator entrypoint.

## Fix — two changes, both small

### A. Make \`git fetch\` non-fatal in resolve-workspace.sh

Same pattern the next line already uses for \`git pull\`:

\`\`\`diff
- git fetch origin
+ git fetch origin 2>/dev/null || true
\`\`\`

This stops the restart loop immediately — the entrypoint continues even if fetch fails, and the orchestrator can come up using whatever's already in the workspace. The user's workflow doesn't get blocked, even if they're temporarily offline or auth is unavailable.

### B. Source \`wizard-credentials.env\` in \`entrypoint-orchestrator.sh\` too

So \`GH_TOKEN\` is actually populated on restart, and \`git fetch\` succeeds normally (pulls latest commits from the user's repo). Mirror the change cluster-base#28 made to post-activation:

\`\`\`diff
  log "Starting orchestrator setup..."

+ # Source wizard-delivered credentials from prior bootstrap (if any).
+ # Same source as entrypoint-post-activation.sh — generacy-ai/generacy#589.
+ WIZARD_CREDS="${WIZARD_CREDS_PATH:-/var/lib/generacy/wizard-credentials.env}"
+ if [ -f "$WIZARD_CREDS" ]; then
+     set -a
+     # shellcheck disable=SC1090
+     source "$WIZARD_CREDS"
+     set +a
+ fi
+
  # Configure git credentials
  bash /usr/local/bin/setup-credentials.sh
\`\`\`

Both fixes are needed:
- **A** alone: prevents the loop but loses the "fetch latest commits on restart" feature (orchestrator boots on stale workspace state)
- **B** alone: works if the wizard creds file exists, but if it doesn't (or the token has expired — installation tokens last ~1h, see #547), \`git fetch\` still fails fatally
- **A + B**: token works when fresh, falls back gracefully when expired

## Caveats / followups

GitHub installation tokens expire after ~1 hour (per #547's "Refresh strategy" note). For long-running clusters that are stopped and restarted hours later, the persisted \`GH_TOKEN\` will be expired. Fix B keeps the path working when fresh; fix A prevents the loop when expired. A long-running token refresh strategy is still needed (track in #547's follow-up).

## Test plan
- [ ] After both fixes: \`docker compose down && up\` on a freshly-bootstrapped cluster no longer restart-loops
- [ ] If the env file exists with a fresh \`GH_TOKEN\`: \`git fetch\` succeeds, latest commits are pulled
- [ ] If the env file doesn't exist OR token is expired: container starts cleanly, log shows the existing \`WARNING: GH_TOKEN not set\` message, orchestrator runs against the current workspace state

## Related
- generacy-ai/generacy#589 (writer side)
- cluster-base#28 (post-activation sourcing — this is the same change for the orchestrator entrypoint)
- generacy-ai/generacy-cloud#547 (token minting; long-running refresh is a follow-up there)
- generacy-ai/generacy#572 (cluster ↔ cloud contract umbrella)

--- a/.devcontainer/generacy/scripts/entrypoint-orchestrator.sh
+++ b/.devcontainer/generacy/scripts/entrypoint-orchestrator.sh
@@ -8,6 +8,21 @@ log() {
 
 log "Starting orchestrator setup..."
 
+# Source wizard-delivered credentials persisted by a prior bootstrap
+# (written by control-plane's bootstrap-complete handler — see
+# generacy-ai/generacy#589). On restarts of an already-bootstrapped
+# cluster, GH_TOKEN lives only in this file; without it, setup-credentials
+# warns and the later `git fetch` in resolve-workspace.sh fails auth.
+# Mirrors the same sourcing block in entrypoint-post-activation.sh.
+WIZARD_CREDS="${WIZARD_CREDS_PATH:-/var/lib/generacy/wizard-credentials.env}"
+if [ -f "$WIZARD_CREDS" ]; then
+    log "Sourcing wizard credentials from $WIZARD_CREDS"
+    set -a
+    # shellcheck disable=SC1090
+    source "$WIZARD_CREDS"
+    set +a
+fi
+
 # Configure git credentials
 bash /usr/local/bin/setup-credentials.sh
 

--- a/.devcontainer/generacy/scripts/resolve-workspace.sh
+++ b/.devcontainer/generacy/scripts/resolve-workspace.sh
@@ -57,6 +57,6 @@ if [ -n "${REPO_URL:-}" ] && [ ! -d "${WORKSPACE_DIR}/.git" ]; then
 elif [ -d "${WORKSPACE_DIR}/.git" ]; then
     log "Project repo already cloned, pulling latest..."
     cd "${WORKSPACE_DIR}"
-    git fetch origin
+    git fetch origin 2>/dev/null || true
     git pull --ff-only origin "${REPO_BRANCH:-main}" 2>/dev/null || true
 fi

